packages/dds/tree/src/util/utils.ts

```diff
@@ -47,18 +47,8 @@ export function asMutable<T>(readonly: T): Mutable<T> {
 
 export const clone = structuredClone;
 
-/**
- * Throw an error with a constant message.
- * @remarks
- * Works like {@link @fluidframework/core-utils/internal#assert}.
- */
-export function fail(message: string | number): never {
-	// Declaring this here aliased to a different name avoids the assert tagging objecting to the usages of `assert` below.
-	// Since users of `fail` do the assert message tagging instead, suppressing tagging errors here makes sense.
-	const assertNoTag: (condition: boolean, message: string | number) => asserts condition =
-		assert;
-	assertNoTag(false, message);
-}
+// TODO: update usages of this to use @fluidframework/core-utils/internal directly.
```

Comment:
There are quite a few usages of this function within tree, so I think splitting this PR and doing those updates as a separate change makes sense, especially given the chance that this change might not be accepted.

Reply:
Agreed, and then there is the further step to find the various places in tree that wanted to use `fail` for convenience, but didn't - because `fail` didn't have assert tagging - and replace those with `fail`.